src/plugins/vis_types/pie/public/pie_component.tsx

```diff
@@ -285,82 +284,111 @@ const PieComponent = (props: PieComponentProps) => {
     ? visData.columns[visParams.dimensions.splitRow[0].accessor]
     : undefined;
 
+  /**
+   * Checks whether data have all zero values.
+   * If so, the no data container is loaded.
+   */
+  const isAllZeros = useMemo(
+    () =>
+      visData.rows.every((row) => {
+        const metricColumnId = metricColumn.id;
+        return row[metricColumnId] === 0;
+      }),
+    [visData.rows, metricColumn]
+  );
+
+  /**
+   * Checks whether data have negative values.
+   * If so, the no data container is loaded.
+   */
+  const hasNegative = useMemo(
+    () =>
+      visData.rows.some((row) => {
+        const value = row[metricColumn.id];
+        return typeof value === 'number' && value < 0;
+      }),
+    [visData.rows, metricColumn]
+  );
```

Comment:
Nice use of every/some! 

It's a style issue but it's nice to inline a variable which has a single use, and the variable name doesn't add significant info to the expression, so instead of 
```
visData.rows.every((row) => {
        const metricColumnId = metricColumn.id;
        return row[metricColumnId] === 0;
      })
```
it can be 
```
visData.rows.every((row) => row[metricColumn.id] === 0)
``` 
or 
```
visData.rows.every(({ [metricColumn.id]: value }) => value === 0)
```